Description: Fix buffer overrun when saving ICO image with a width that's not a multiple of 4
Author: DS
Applied-Upstream: http://trac.wxwidgets.org/changeset/67297
Last-Update: 2012-05-31

Index: wxWidgets/branches/WX_2_8_BRANCH/src/common/imagbmp.cpp
===================================================================
--- a/src/common/imagbmp.cpp (revision 54942)
+++ b/src/common/imagbmp.cpp (revision 67297)
@@ -137,6 +137,6 @@
 
     unsigned width = image->GetWidth();
-    unsigned row_padding = (4 - int(width*bpp/8.0) % 4) % 4; // # bytes to pad to dword
-    unsigned row_width = int(width * bpp/8.0) + row_padding; // # of bytes per row
+    unsigned row_padding = (4 - ((width * bpp + 7) / 8) % 4) % 4; // # bytes to pad to dword
+    unsigned row_width = (width * bpp + 7) / 8 + row_padding; // # of bytes per row
 
     struct
Index: wxWidgets/branches/WX_2_8_BRANCH/docs/changes.txt
===================================================================
--- a/docs/changes.txt (revision 67259)
+++ b/docs/changes.txt (revision 67297)
@@ -132,4 +132,5 @@
 - Fixed compilation with libpng 1.5 and above.
 - Removed ugly black (simple) border around wxHTML help window.
+- Fixed buffer overrun when saving ICO image with a width that's not a multiple of 4.
 
 wxMSW:
